base/random.jl

```diff
@@ -1203,7 +1203,7 @@ let Floats = Union{Float16,Float32,Float64}
             $randfun{T<:$Floats}(::Type{T}) = $randfun(GLOBAL_RNG, T)
 
             # filling arrays
-            function $randfun!{T}(rng::AbstractRNG, A::AbstractArray{T})
+            function $randfun!{T<:$Floats}(rng::AbstractRNG, A::AbstractArray{T})
```

Comment:
Would it work to restrict less tightly with `T<:Float` ? We can imagine that a custom RNG implements e.g. `randn(customRNG, BigFloat)`, and then wants to use the fallback `randn!(customRNG, BigFloat[1.0])`.